### PR TITLE
Support user dictionary loading

### LINKED ISSUE #488: [v1.4] Support user-dictionary loading
*Following this [central issue](https://github.com/meilisearch/integration-guides/issues/286)*

- [ ]  Add a new setting: `dictionary` with `get`, `update`, and `reset` methods associated.
Here are the JS equivalents in meilisearch-js you should create for this repository:

```jsx
client.index('indexName').getDictionary(); // calls GET /indexes/:uid/settings/dictionary
client.index('indexName').updateDictionary(['|', '/', '&sep']);  // calls PUT /indexes/:uid/settings/dictionary
client.index('indexName').resetDictionary(); // calls DELETE /indexes/:uid/settings/dictionary
```

- [ ]  The methods associated with the `/settings` API route (to get, reset and update the settings globally) must be able to receive in the payload the new configuration `dictionary`.
- [ ]  ⚠️ Add integration tests ⚠️
- [ ]  Update the `.code-samples.meilisearch.yaml`
    - [ ]  Add `get_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1133
    - [ ]  Add `update_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1136
    - [ ]  Add `reset_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1144

### PR DESCRIPTION
**Pull request by ellnix**
# Pull Request

## Related issue
Fixes #488 

## PR checklist
Please check if your PR fulfills the following requirements:
- [X] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [X] Have you read the contributing guidelines?
- [X] Have you made sure that the title is accurate and descriptive of the changes?

I attempted to keep to the repo's style and make as few edits as possible.

### CODE REVIEW

File: lib/meilisearch/index.rb [hunk 1]
```diff
@@ -461,5 +461,22 @@ def update_faceting(faceting_attributes)
     def reset_faceting
       http_delete("/indexes/#{@uid}/settings/faceting")
     end
+
+    ### SETTINGS - DICTIONARY
+
+    def dictionary
+      http_get("/indexes/#{@uid}/settings/dictionary")
+    end
+    alias get_dictionary dictionary
```
Comment (brunoocasali) [inline lib/meilisearch/index.rb:470]: Same thing as the other PR, let's remove these aliases :)

Comment (ellnix) [reply lib/meilisearch/index.rb:470]: Same as the other PR, I am assuming you mean the getter only, and leaving the setter alias.

**Comment by brunoocasali** [comment]
Can you fix the conflicts here?

File: lib/meilisearch/index.rb [hunk 2]
```diff
@@ -461,5 +461,21 @@ def update_faceting(faceting_attributes)
...
+
+    def dictionary
+      http_get("/indexes/#{@uid}/settings/dictionary")
+    end
+
+    def update_dictionary(dictionary_attributes)
+      attributes = Utils.transform_attributes(dictionary_attributes)
+      http_put("/indexes/#{@uid}/settings/dictionary", attributes)
+    end
+    alias dictionary= update_dictionary
```
Comment (brunoocasali) [inline lib/meilisearch/index.rb:475]:
```suggestion
```

**Comment by ellnix** [comment]
> Can you fix the conflicts here?

Done. Rebased and force pushed, everything should be clean. Let me know if something else is required.

**Review by brunoocasali — DISMISSED** [review]
bors merge

**Comment by ellnix** [comment]
@brunoocasali 
The two linter errors are that 
- `class Index` is too long and 
- the settings spec block is too long. 

I do not feel like these are problems that can be really addressed in this PR, perhaps some refactoring should be done?

**Comment by brunoocasali** [comment]
I think we can just regenerate the rubocop todo file, and then address those issues in a new PR :)

**Comment by ellnix** [comment]
Noted, will do first thing tomorrow.

**Comment by ellnix** [comment]
@brunoocasali all done & rebased, should be good to go with 1 click. Sorry for being slow on the catch up, I wasn't familiar with the rubocop auto gen workflow. I will hang around to rebase the other PRs when this one gets merged. Thanks for the patience.

**Comment by brunoocasali** [comment]
I should thank you. In fact, you're doing an amazing job here!

**Review by brunoocasali — APPROVED** [review]
bors merge

**Pull request closed without merging**